### PR TITLE
Adding delete_host method and get_workers nodes

### PR DESCRIPTION
**Pull request by lalon4**

### CODE REVIEW

File: discovery-infra/test_infra/assisted_service_api.py [hunk 1]
```diff
@@ -189,6 +189,10 @@ def delete_cluster(self, cluster_id):
         log.info("Deleting cluster %s", cluster_id)
         self.client.deregister_cluster(cluster_id=cluster_id)
 
+    def delete_host(self, cluster_id, host_id):
```
Comment (nshidlin) [inline discovery-infra/test_infra/assisted_service_api.py:192]: Maybe should be `deregister_host`? wdty?

Comment (lalon4) [reply discovery-infra/test_infra/assisted_service_api.py:192]: isn't delete_host is more intuitive when:
new_cluster.delete_host(host)?
i mean, youll know to search deregister if you know that the API called deregister rather then delete
WDYT?

Comment (lalon4) [reply discovery-infra/test_infra/assisted_service_api.py:192]: ho its the api, so yeah it can be deregistered, changing

**Pull request merged**